src/test/kotlin/org/taktik/icure/asynclogic/GenericLogicImplTest.kt

```diff
@@ -0,0 +1,59 @@
+package org.taktik.icure.asynclogic
+
+import java.lang.IllegalStateException
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.flow.count
+import org.junit.jupiter.api.TestInstance
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.taktik.icure.test.FakeDto
+import org.taktik.icure.test.FakeLogicImpl
+import org.taktik.icure.test.ICureTestApplication
+
+private const val TEST_CACHE = "build/tests/icureCache"
+
+@SpringBootTest(
+	classes = [ICureTestApplication::class],
+	properties = [
+		"spring.main.allow-bean-definition-overriding=true",
+		"icure.objectstorage.icureCloudUrl=test",
+		"icure.objectstorage.cacheLocation=$TEST_CACHE",
+		"icure.objectstorage.backlogToObjectStorage=true",
+		"icure.objectstorage.sizeLimit=1000",
+		"icure.objectstorage.migrationDelayMs=1000",
+	],
+	webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ActiveProfiles("app")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class GenericLogicImplTest (
+	fakeLogicImpl: FakeLogicImpl
+): StringSpec({
+
+	"Can create some DTOs if all of them have the rev field equal to none" {
+		fakeLogicImpl.fakeDAO.resetCache()
+		val entities = listOf(
+			FakeDto(id = "FAKE1", rev = null, revHistory = null),
```

Comment:
In order to avoid creating some objects only for tests, can't you use the existing ones ? (PatientDAOImpl, PatientDto, ...)

Reply:
For example, you can : 
- In your test class, create a "FakeLogic", implementing GenericLogicImpl; 
- mock an existing class (https://mockk.io/#class-mock) to send back a response to the genericDAO() method; 
- mock create() method of the mock above, in case it is a success. Otherwise, your test will treat the error; 
That way, you can keep everything inside your test class, without adding new objects dedicated only for those tests